### ISSUE TITLE
Export notes to PDF

### ISSUE THREAD

**Issue by hasanen** [standardnotes/forum#27]
I just migrated rest of my notes to Standard Notes on this week and I've already had couple of situations where it would have been handy to be able to export note to PDF.

I'm on macOS and currently I copy paste note to iA Writer to print notes to PDF's.

And in future I'm quite sure I'd like to have similar functionality in Android/iOs apps too. 

Is this something that we could get in near future? 

And thanks for the good work with the app, I finally found somethign to replace my vimwiki-notebook :)

**Comment by hasanen** [comment]
Created an issue https://github.com/standardnotes/web/issues/143

**Comment by mobitar** [comment]
Ya, this sounds reasonable. Would come in the form of an extension, probably as part of the [action bar](https://standardnotes.org/extensions/action-bar). This is a better place for this than the web repo, so I'll reopen it here.

**Comment by sergeydgr8** [comment]
+1 to exporting notes to PDF. Frequently find myself wanting to share something but I have to use another app to do so.

Speaking of sharing, it would be nice to be able to get shareable links to notes. Not sure if an issue has been opened for that yet.

**Comment by hasanen** [comment]
@sergeydgr8 did you already check this https://standardnotes.org/extensions/share?

**Comment by sergeydgr8** [comment]
@hasanen yep, was already pointed to it in a different thread. Neat stuff!

**Comment by harrisondesbrosses** [comment]
@hasanen I'm curious where your original link led to? Were there some older share extensions that don't exist anymore? I'm definitely interested in saving notes to PDF.

**Comment by harrisondesbrosses** [comment]
Ah nevermind, it was a shareable link mention - any update on a PDF export ability?

**Comment by mobitar** [comment]
This is definitely out of scope for us. I'm surprised I said this was reasonable (I was probably excited by having launched Action Bar), but I doubt we'll add this. It's just not something we can probably be very good at, so it's best to leave this speciality behavior to applications that specialize in it and give users everything they ever wanted in a PDF generation app :)

**Comment by jrabensc** [comment]
Maybe this could be solved using a customizable action button in the action bar to call for example pandoc and create the pdf via the shell. What do you think? Currently I am exporting my .md-files, switch to the shell and run the command manually.

**Comment by mobitar** [comment]
Possibly, but the user would need to have pandoc installed. In that case, the extension we create should have an option for user to create their own links or actions. But that's a whole other thing..

**Comment by jrabensc** [comment]
> the extension we create should have an option for user to create their own links or actions.

This would be awesome!

**Comment by luisborba** [comment]
Any recommendations for a file converter for Mac that has  GUI? I installed Pandoc and my head hurts now trying to use the Terminal.

**Comment by simgooder** [comment]
Similar thread to the _save as PDF_ discussion, but would it be a lot to ask for a _copy as..._ function in the action-bar? 
For instance, if I copy/paste the markdown preview code, it's only copied as plain-text. If I hit _copy_ from the action bar, it's copied as markdown. It would be massively beneficial to my workflow to be able to be able to _copy as..._ _rich-text_ or _html_.

**Comment by mobitar** [comment]
An extension that translates between different formats could be cool, in general. Not sure we would squeeze that into Action Bar. The LaTeX editor does have this option built in via the tabs at the top, and allows you to view the raw HTML behind the Markdown.

**Comment by jonny-379** [comment]
How is it that five years after the original PDF request, there is still no export as PDF function...

I just started to use standard notes for university notes and use the "Bold Editor" so I can add pictures, but I noticed a few days ago that if trying to export notes as files from the advanced editor it is only as HTML, and without pictures... 

I think this is **really** bad, now I can not export and share my notes with other students (well I can share notes but without any images or with an absurd amount of work to download and add the images. Or I could print in the web app and have the GUI element on every page)... I was about to recommend the program to others, but without a basic "export as pdf" function is this program for a lot of users useless...

Don't get me wrong, overall I like the program and I'm going to continue to use the program (mainly because I paid for it), I like the privacy aspect and started to use it also for personal notes.

**Comment by danielnetop** [comment]
I totally agree, I like standard notes as a product, but it lacks what I would call a basic function (not saying is basic to implement, but the functionality itself should be a basic export on a note), and this is the reason why I still have Joplin installed on all machines I use standard notes, it exports to pdf without any issues, I've used directly pandoc but it's definitely not the same, but I do agree that it's been quite some time without the feature to export to pdf from the markdown editors.
